conda_smithy/configure_feedstock.py

```diff
@@ -95,7 +112,19 @@ def warn_once(msg: str):
     logger.warning(msg)
 
 
-def package_key(config, used_loop_vars, subdir):
+def package_key(
+    config: Dict[
+        str,
+        Union[
+            List[str],
+            OrderedDict,
+            List[Union[str, List[str]]],
+            List[List[str]],
+        ],
```

Comment:
I am not familiar with mypy specifically, so this is a hunch based on other type systems: I would expect the type `List[Union[str, List[str]]]` to be identical to `Union[List[str], List[List[str]]]`, so can this whole type be reduced to `Union[List[Union[str, List[str]]], OrderedDict]`?

Reply:
Or, if the nested unions are too confusing, `Union[List[str], List[List[str]], OrderedDict]`?